src/parsing.jl

```diff
@@ -227,14 +227,17 @@ function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:timesta
         return ZonedDateTime(typemin(DateTime), tz"UTC")
     end
 
+    # Cut off digits after the third after the decimal point,
+    # since DateTime in Julia currently handles only milliseconds, see Issue #33
+    str = replace(str, r"(\.[\d]{3})\d+", s"\g<1>")
```

Comment:
Should try parsing with the first three formats, then run replacement, then return the result from parsing the last format.

Reply:
ok, so I just moved the replacement after the loop, seems to work :-)

Reply:
Unfortunately the tests won't hit that case since TimeZones had a bug when I wrote this. I'll revisit it soon to see if I can enable those tests. But this is good for now!